src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractImportCompletionProvider.cs

```diff
@@ -109,26 +130,32 @@ internal override async Task<CompletionChange> GetChangeAsync(Document document,
             LogCommit();
             var containingNamespace = ImportCompletionItem.GetContainingNamespace(completionItem);
 
+            var provideParenthesisCompletion = commitKey == ';' && ImportCompletionItem.GetProvideParenthesisCompletion(completionItem);
```

Comment:
This code seems to be repeated in each provider, let's put it in a helper class and reuse.